### PR TITLE
Remove cluster view on cluster remove

### LINKED ISSUE #756: Cluster dashboard iframe is not removed from DOM on cluster remove

### PR DESCRIPTION
**Pull request by nevalla**
Fixes #756 

Signed-off-by: Lauri Nevala <lauri.nevala@gmail.com>

### CODE REVIEW

File: src/renderer/components/cluster-manager/clusters-menu.tsx [hunk 1]
```diff
@@ -76,8 +77,11 @@ export class ClustersMenu extends React.Component<Props> {
             label: _i18n._(t`Remove`),
           },
           ok: () => {
+            if (!clusterStore.activeClusterId || clusterStore.activeClusterId === cluster.id) {
```
Comment (ixrock) [inline src/renderer/components/cluster-manager/clusters-menu.tsx]: Btw, this is fixed in #740

File: src/renderer/components/cluster-manager/lens-views.ts [hunk 2]
```diff
@@ -34,6 +34,17 @@ export async function initView(clusterId: ClusterId) {
   parentElem.appendChild(iframe);
 }
 
+export async function removeView(clusterId: ClusterId) {
```
Comment (ixrock) [inline src/renderer/components/cluster-manager/lens-views.ts]: Adding this to the end of `initView` should be enough (without tweaking other files)
```ts
  // auto-clean when cluster removed
  await when(() => !clusterStore.getById(clusterId));
  parentElem.removeChild(iframe)
  lensViews.delete(clusterId)
```

**Review by ixrock — CHANGES_REQUESTED** [review]

File: src/renderer/components/cluster-manager/lens-views.ts (hunk 2 above)
Comment (nevalla) [reply src/renderer/components/cluster-manager/lens-views.ts]: Thanks...testing it

**Comment by nevalla** [comment]
@ixrock PTAL again

**Review by ixrock — APPROVED** [review]

**Pull request merged**